cmd/swarm/swarm-smoke/main.go

```diff
@@ -110,10 +141,36 @@ func main() {
 
 	sort.Sort(cli.FlagsByName(app.Flags))
 	sort.Sort(cli.CommandsByName(app.Commands))
+	app.Before = func(ctx *cli.Context) error {
+		return nil
+	}
+	app.After = func(ctx *cli.Context) error {
+		emitMetrics(ctx)
+		return nil
+	}
 
 	err := app.Run(os.Args)
 	if err != nil {
 		log.Error(err.Error())
+
 		os.Exit(1)
 	}
 }
+
+func emitMetrics(ctx *cli.Context) {
+	if gethmetrics.Enabled {
+		var (
+			endpoint = ctx.GlobalString(swarmmetrics.MetricsInfluxDBEndpointFlag.Name)
+			database = ctx.GlobalString(swarmmetrics.MetricsInfluxDBDatabaseFlag.Name)
+			username = ctx.GlobalString(swarmmetrics.MetricsInfluxDBUsernameFlag.Name)
+			password = ctx.GlobalString(swarmmetrics.MetricsInfluxDBPasswordFlag.Name)
+			hosttag  = ctx.GlobalString(swarmmetrics.MetricsInfluxDBHostTagFlag.Name)
+		)
+
```

Comment:
I myself would validate that all of these variables have `length > 0`

Reply:
i'm not positive if all of them should have `length>0`. password can be `""`, username probably also, host tag too, database not sure but might. endpoint maybe but its pretty obvious that there should be a string there :) let's leave as is

Reply:
i don't think we have to validate this input - if it is wrong, you notice because your measurements are missing.